spec/app/ics-029-fee-payment/README.md

```diff
@@ -134,13 +137,18 @@ function acknowledgePacket(
   proof: CommitmentProof,
   proofHeight: Height,
   relayer: string): Packet {
-    // get the forward relayer from the acknowledgement
-    // and pay fees to forward and reverse relayers.
-    // reverse_relayer is submitter of acknowledgement message
-    // provided in function arguments
-    // NOTE: Fee may be zero
-    forward_relayer = getForwardRelayer(acknowledgement)
-    PayFee(packet, forward_relayer, relayer)
+    // get the underlying connection for this channel
+    connection = getConnection()
+    // only call the fee module callbacks if the connection supports incentivization.
+    if IsSupportedFeature(connection, "INCENTIVE_V1") {
+        // get the forward relayer from the acknowledgement
+        // and pay fees to forward and reverse relayers.
+        // reverse_relayer is submitter of acknowledgement message
+        // provided in function arguments
+        // NOTE: Fee may be zero
+        forward_relayer = getForwardRelayer(acknowledgement)
+        PayFee(packet, forward_relayer, relayer)
```

Comment:
I brought up something similar in #582 

It is good you define the interface between ICS-4 and the `PayFee` fee handler.

However, it doesn't seem like the call from ICS-4 to the application is specified anywhere. This seems a bit asymetric. How should we add that?

Reply:
Added, let me know if you had something else in mind